pkgs/by-name/ya/yafu/package.nix

```diff
@@ -0,0 +1,147 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  makeWrapper,
+  gnum4,
+  gmp,
+  ecm,
+  enableCuda ? false,
+  enableSse41 ? false,
+  enableAvx2 ? false,
+  enableBmi2 ? false,
+  enableSkylakex ? false,
+  enableIcelake ? false,
+  disableZlib ? false,
+  vbits ? null,
+  cudaPackages ? null,
+  zlib,
+}:
+
+assert enableCuda -> cudaPackages != null;
+assert
+  vbits != null
+  -> lib.elem vbits [
+    64
+    128
+    256
+  ];
+
+stdenv.mkDerivation {
+  pname = "yafu";
+  version = "3.0-unstable-2024-11-04";
+
+  src = fetchFromGitHub {
+    owner = "bbuhrow";
+    repo = "yafu";
+    rev = "ada8bf3b43a684d6217891f593fd7c6f5e61982d";
+    hash = "sha256-w6BhDNQu5GxOwFRqHXBEsXN1UwaMfeAVvLeRfy4QndE=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+    gnum4
+  ];
+
+  buildInputs = [
+
+    (gmp.override { withStatic = true; })
+  ]
+  ++ lib.optional enableCuda cudaPackages.cuda_cudart
+  ++ lib.optional (!disableZlib) zlib;
+
+  # YAFU's build system doesn't have a configure phase
+  dontConfigure = true;
+
+  makeFlags =
+    let
+      hp = stdenv.hostPlatform;
+    in
+    lib.optionals enableCuda [ "CUDA=1" ]
+    ++ lib.optionals disableZlib [ "NO_ZLIB=1" ]
+    ++ lib.optionals (vbits != null) [ "VBITS=${toString vbits}" ]
+    ++ lib.optionals (hp.sse4_1Support or enableSse41) [ "USE_SSE41=1" ]
+    ++ lib.optionals (hp.avx2Support or enableAvx2) [ "USE_AVX2=1" ]
+    ++ lib.optionals (hp.avx2Support or enableBmi2) [ "USE_BMI2=1" ]
+    ++ lib.optionals (hp.avx512Support or enableSkylakex) [ "SKYLAKEX=1" ]
+    ++ lib.optionals ((lib.hasPrefix "icelake" (hp.gcc.arch or "")) || enableIcelake) [ "ICELAKE=1" ];
+
+  env.NIX_CFLAGS_COMPILE = lib.concatStringsSep " " [
+    "-Wno-error=int-conversion"
+    "-Wno-error=incompatible-pointer-types"
+    "-Wno-error=implicit-function-declaration"
+  ];
+
+  postPatch = ''
+    sed -i 's|^CFLAGS=.*-DBITS_PER_GMP_ULONG=32.*|CFLAGS=-m64 -g -Ofast -march=native -I. -I${
+      gmp.override { withStatic = true; }
+    }/include -DULL_NO_UL -DBITS_PER_GMP_ULONG=64|' factor/lasieve5_64/Makefile
+
+    sed -i '/-I.*mingw.*include/d' factor/lasieve5_64/Makefile
```

Comment:
(Personal preference; Applicable for all multi seds in the package) When you have multiple `sed`s on the same file you can instead do:
   ```
   sed -i \
    -e PATTERN \
    -e PATTER \
    FILE
   ```
(You may resolve this without addressing it)